packages/server/src/users.ts

```diff
@@ -14,6 +14,25 @@ export interface PersistentUser extends UserResponse {
   username: string;
   password_hash: string;
   login_type: "persistent";
+  rating?: number;
+}
+
+export async function updateUserRating(id: string, new_rating: number): Promise<void> {
+  if((await getUserBySessionId(id)).login_type == 'guest'){
+    throw new Error("Guest users cannot have ratings");
```

Comment:
You are right that enforcing this would add an additional query and slow things down. I just thought that since guests are temporary users, updating a rating for guests would be trivial.